beginner/irods_beginner_training_2024.tex

```diff
@@ -1505,6 +1506,104 @@ \section{Microservices}
 }
 \end{lstlisting}
 
+\section{Using the Delay Queue}
+\label{sec:delaydemo}
+
+To demonstrate use of the delay queue, we will create a rule that utilizes \texttt{delay()}, submit it to the delay queue, and verify its output in the system log.
+
+First, become the \texttt{rods} user.
+\begin{lstlisting}
+$ iexit full
+\end{lstlisting}
+
+\begin{lstlisting}
+$ rm ~/.irods/irods_environment.json
+\end{lstlisting}
+\pagebreak
+Execute \texttt{iinit}.
+
+On the same line, after the colon, enter: \texttt{X.X.X.X}
+
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily]
+$ iinit
+One or more fields in your iRODS environment file (irods_environment.json) are missing; please enter them.
+Enter the host name (DNS) of the server to connect to:
+\end{lstlisting}
+
+On the same line, after the colon, enter: \texttt{1247}
+
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily]
+Enter the port number:
+\end{lstlisting}
+
+On the same line, after the colon, enter: \texttt{rods}
+
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily]
+Enter your irods user name:
+\end{lstlisting}
+
+On the same line, after the colon, enter: \texttt{tempZone}
+
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily]
+Enter your irods zone:
+\end{lstlisting}
+
+Remember we set the rods password as \texttt{rods}.
+
+\begin{lstlisting}[basicstyle=\scriptsize\ttfamily]
+Those values will be added to your environment file (for use by
+other iCommands) if the login succeeds.
+
+Enter your current iRODS password:
+\end{lstlisting}
+
+Next, create a file named \texttt{example\_delay.r} with the following content:
+
+\begin{lrbox}{\exampleDelayRule}
+\lstinputlisting[basicstyle=\scriptsize\ttfamily]{example_delay.r}
+\end{lrbox}
+\href{https://raw.githubusercontent.com/irods/irods_training/ugm2024/beginner/example_delay.r}{\usebox{\exampleDelayRule}}
+
+This rule will print the string \texttt{Adding delay rule to queue.} to the server log. After 30 seconds, this rule will print the string \texttt{Hello iRODS World!} to the server log, and then every minute thereafter.
+
+To submit this rule file:
+
+\begin{lstlisting}
+$ irule -r irods_rule_engine_plugin-irods_rule_language-instance -F example_delay.r
+\end{lstlisting}
+
+To verify that the rule has been added to the delay queue:
+
+\begin{lstlisting}
+$ iqstat
+\end{lstlisting}
+
+The output should resemble:
+\begin{lstlisting}
+id     name
+10122  
+                writeLine("serverLog", "Hello iRODS world!"); 
+\end{lstlisting}
+
+Next, verify that the rule is successfully running by checking the system log:
+
+\begin{lstlisting}
+$ sudo journalctl -f --no-pager _COMM=irodsServer
```

Comment:
On my VM, when I'm at the ending state after doing the beginner training, it seems like rsyslog is the default (?). I couldn't find `/var/log/irods`, but everything popped out of `journalctl` as expected.

Reply:
Oh, forgot to answer your first question. It prints out system log entries without a pager (so no `less`, `more`, etc. and the whole thing is wordwrapped so you can actually see it) and follows it (`-f`), so like `tail -f`. `_COMM=irodsServer` only filters out log entries where the process name is `irodsServer`.

Reply:
If someone is doing this on their personal computer and they use something other than rsyslog, then that's a vote to leave this as is. Rsyslog isn't a requirement of iRODS either, so that's a second vote.

I'm starting to lean in the direction of keeping what you have as is. My only question is ... is `journalctl` always available on the platforms we support?

Reply:
I'm not sure. Please do a little googling to verify.

Reply:
I checked, and I think all of them do use `systemd`. Would like verification from somewhere, though.

Reply:
centos7 is the oldest supported - if it's fine - i think we're okay.